src/HealthGPS/dynamic_hierarchical_linear_model.cpp

```diff
@@ -25,8 +29,17 @@ RiskFactorModelType DynamicHierarchicalLinearModel::type() const noexcept {
 
 std::string DynamicHierarchicalLinearModel::name() const noexcept { return "Dynamic"; }
 
-void DynamicHierarchicalLinearModel::generate_risk_factors(
-    [[maybe_unused]] RuntimeContext &context) {}
+void DynamicHierarchicalLinearModel::generate_risk_factors(RuntimeContext &context) {
+
+    // Adjust risk factors such that mean sim value matches expected value.
+    std::vector<core::Identifier> factor_keys;
+    factor_keys.reserve(variables_.size());
+    for (const auto &factor : variables_) {
+        // factor.second contains the factor name.
+        factor_keys.emplace_back(factor.second);
+    }
+    adjust_risk_factors(context, factor_keys);
+}
```

Comment:
Does this mean that any `DynamicHierarchicalLinearModel` will want to adjust the risk factors, always? For some reason, I was under the impression that some models will want to adjust the risk factors while others will not.

Reply:
Yes, your impression is correct. In future, once generic modules #231 is implemented, I would like to separate baseline adjustments out completely into it's own module, where a user would have one or more of these in their module ordered list (for timing), with their own variables they operate on. Time and existing code constraints mean I need to squash it in here for now though.